x/rollup/keeper/msg_server.go

```diff
@@ -230,6 +195,61 @@ func (k *Keeper) SetL1BlockHistory(ctx context.Context, info *derive.L1BlockInfo
 	return nil
 }
 
+// processL1SystemDepositTx processes the L1 system deposit tx and returns the L1 block info.
+func (k *Keeper) processL1SystemDepositTx(ctx sdk.Context, txBytes []byte) (*derive.L1BlockInfo, error) {
+	var tx ethtypes.Transaction
+	if err := tx.UnmarshalBinary(txBytes); err != nil {
+		ctx.Logger().Error("Failed to unmarshal system deposit transaction", "index", 0, "err", err, "txBytes", txBytes)
+		return nil, types.WrapError(types.ErrInvalidL1Txs, "failed to unmarshal system deposit transaction: %v", err)
+	}
+	if !tx.IsDepositTx() {
+		ctx.Logger().Error("First L1 tx must be a system deposit tx", "type", tx.Type())
+		return nil, types.WrapError(types.ErrInvalidL1Txs, "first L1 tx must be a system deposit tx, but got type %d", tx.Type())
+	}
+	l1blockInfo, err := derive.L1BlockInfoFromBytes(k.rollupCfg, 0, tx.Data())
+	if err != nil {
+		ctx.Logger().Error("Failed to derive L1 block info from L1 Info Deposit tx", "err", err, "txBytes", txBytes)
+		return nil, types.WrapError(types.ErrInvalidL1Txs, "failed to derive L1 block info from L1 Info Deposit tx: %v", err)
+	}
+	return l1blockInfo, nil
+}
+
+// processL1UserDepositTxs processes the L1 user deposit txs and mints ETH to the user's cosmos address.
+func (k *Keeper) processL1UserDepositTxs(ctx sdk.Context, txs [][]byte) error {
+	for i := 1; i < len(txs); i++ {
+		txBytes := txs[i]
+		var tx ethtypes.Transaction
+		if err := tx.UnmarshalBinary(txBytes); err != nil {
+			ctx.Logger().Error("Failed to unmarshal user deposit transaction", "index", i, "err", err, "txBytes", txBytes)
+			return types.WrapError(types.ErrInvalidL1Txs, "failed to unmarshal user deposit transaction", "index", i, "err", err)
+		}
+		if !tx.IsDepositTx() {
+			ctx.Logger().Error("L1 tx must be a user deposit tx", "index", i, "type", tx.Type())
+			return types.WrapError(types.ErrInvalidL1Txs, "L1 tx must be a user deposit tx, index:%d, type:%d", i, tx.Type())
+		}
+		if tx.IsSystemTx() {
```

Comment:
Out of scope for this PR, but our usage here of `IsSystemTx` probably isn't current.

https://specs.optimism.io/protocol/regolith/overview.html

Reply:
Good catch! I'll make a github issue to investigate/update this

Reply:
https://github.com/polymerdao/monomer/issues/128